{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Abstract\n",
    "\n",
    "We find histologies associated with tissues containing NF1 somatic mutations in the COSMIC database. There are 21. We find all other gene somatic mutations for those histologies.  We assess the main predisposing gene to be the gene with the most mutation-sample pairs associated with the histology.  We find that 4 histologies are primarily associated with NF1.  We find that the mutation sites in NF1 observed to the 4 histologies each are associated with only a single histology. For MPNST, 3 genes are equally scored and Neurofibroma has a very high score for NF1. We find that each histology is associated with multiple mutation sites. For the NF1 mutation sites associated with each histology, we give FATHM score and PUBMED ID."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "NOTE: All of the samples are taken from people with somatic mutations leading to cancer, so there is selection bias in the samples to that extent.\n",
    "\n",
    "TODO: Program-based search for variant type, affected pathways and clinical significance from NCBI."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Required packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os, re, time\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:100% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.core.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:100% !important; }</style>\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Getting the COSMIC database of genome screens\n",
    "\n",
    "For development purposes we have downloaded the COSMIC database from their website.  We have also uploaded it as a ZIP file to private file syn20716008.  It has license and sharing conditions and is 10GB in size, which makes it difficult to Dockerize.  Let's assume you have downloaded the file and placed it in a folder, and that there is an environment variable NF2_DATA that points to that folder.  This file, uncompressed, takes about an hour to read on a laptop PC with 16GB of RAM and 45GB of paging space.  It takes about 10 minutes to read on a workstation with 32GB of RAM, Intel core i9 processor and solid state disk."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_dir=os.getenv('NF2_DATA')\n",
    "df=pd.read_csv(f'{data_dir}/CosmicGenomeScreensMutantExport.tsv', sep='\\t')\n",
    "df=df.loc[:,['Gene name','Mutation CDS', 'ID_sample', 'Primary histology']].dropna()\n",
    "df.columns=['gene', 'mutation', 'sampleid', 'histology']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### We find histologies associated with tissues containing NF1 somatic mutations in the COSMIC database"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Remove Not Sure histologies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "df=df[df.histology != 'other']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "df=df[df.histology != 'NS']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Make another dataframe for just NF1 mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "nf1=df[df.gene=='NF1']\n",
    "dd=nf1.to_dict()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "nf1_histologies=set([y for x,y in dd['histology'].items()])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### There are 21"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(nf1_histologies)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### We find all other gene somatic mutations for those histologies"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get the list of tissue samples studied for NF1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "nf1_samples=set([y for x,y in dd['sampleid'].items()])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Restrict the set of all mutations to those concerning NF1-related histologies and samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "def strip_gene(x):\n",
    "    return x.split('_')[0] if '_' in x else x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_nf1=df[df.histology.isin(nf1_histologies)]\n",
    "df_nf1=df_nf1[df_nf1.sampleid.isin(nf1_samples)]\n",
    "genes=df_nf1.gene.unique()\n",
    "samples=df_nf1.sampleid.unique()\n",
    "histologies=df_nf1.histology.unique()\n",
    "M=df_nf1.values\n",
    "f=np.vectorize(strip_gene)\n",
    "M[:,0]=f(M[:,0])\n",
    "W={histology:{} for gene, mutation, sample, histology in M}\n",
    "for gene, mutation, sample, histology in M:\n",
    "    W[histology][gene]={}\n",
    "for gene, mutation, sample, histology in M:\n",
    "    W[histology][gene][mutation]=set([])\n",
    "for gene, mutation, sample, histology in M:\n",
    "    W[histology][gene][mutation].add(sample)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### We assess the main predisposing gene for each histology"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Our scoring function is the gene with the most mutation-sample pairs associated with the histology"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "n=3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [],
   "source": [
    "def score(mutation_samples):\n",
    "    return sum([len(mutation_samples[x]) for x in mutation_samples])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "causative = []\n",
    "not_causative = []\n",
    "for histology in W:\n",
    "    TT=list(reversed(sorted([(score(W[histology][gene]),gene) for gene in W[histology]])))\n",
    "    total_mutations=sum([x for x,y in TT])\n",
    "    TT1=[(y,x/total_mutations) for x,y in TT]\n",
    "    genes=[x for x,y in TT1]\n",
    "    if 'NF1' in genes[0:n]:\n",
    "        i=[i for i,x in enumerate(genes[0:20]) if x=='NF1'][0]\n",
    "        if genes[0]=='NF1' or TT1[0][1]==TT1[i][1]:\n",
    "            causative.append((histology, [x for x in TT1 if x[1]==TT1[0][1]]))\n",
    "        else:\n",
    "            not_causative.append(histology)\n",
    "    else:\n",
    "        not_causative.append(histology)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 17 histologies do not score as causing NF1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Not caused by NF1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>lymphoid_neoplasm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>haematopoietic_neoplasm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>malignant_melanoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>glioma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>carcinoid-endocrine_tumour</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>angiosarcoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>adenoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>primitive_neuroectodermal_tumour-medulloblastoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Ewings_sarcoma-peripheral_primitive_neuroectod...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>osteosarcoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>rhabdomyosarcoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>fibroepithelial_neoplasm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>germ_cell_tumour</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>low_malignant_potential_borderline_tumour</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Wilms_tumour</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>pulmonary_blastoma</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Not caused by NF1\n",
       "0                                           carcinoma\n",
       "1                                   lymphoid_neoplasm\n",
       "2                             haematopoietic_neoplasm\n",
       "3                                  malignant_melanoma\n",
       "4                                              glioma\n",
       "5                          carcinoid-endocrine_tumour\n",
       "6                                        angiosarcoma\n",
       "7                                             adenoma\n",
       "8    primitive_neuroectodermal_tumour-medulloblastoma\n",
       "9   Ewings_sarcoma-peripheral_primitive_neuroectod...\n",
       "10                                       osteosarcoma\n",
       "11                                   rhabdomyosarcoma\n",
       "12                           fibroepithelial_neoplasm\n",
       "13                                   germ_cell_tumour\n",
       "14          low_malignant_potential_borderline_tumour\n",
       "15                                       Wilms_tumour\n",
       "16                                 pulmonary_blastoma"
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame(not_causative, columns=['Not caused by NF1'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### We find that 4 histologies are scored as causative for  NF1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [],
   "source": [
    "rows=[]\n",
    "for histo,L in causative:\n",
    "    for (gene, score) in L:\n",
    "        rows.append((score, gene, histo))\n",
    "rows.sort()\n",
    "rows=list(reversed(rows))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>histology</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>pheochromocytoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>neuroblastoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>malignant_peripheral_nerve_sheath_tumour</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>neurofibroma</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  histology\n",
       "0                          pheochromocytoma\n",
       "1                             neuroblastoma\n",
       "2  malignant_peripheral_nerve_sheath_tumour\n",
       "3                              neurofibroma"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nf1_histologies=[x for x,y in causative]\n",
    "pd.DataFrame(nf1_histologies, columns=['histology'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4"
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_nf1_histologies=len(nf1_histologies)\n",
    "n_nf1_histologies"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### For MPNST, 3 genes are equally scored and Neurofibroma has a very high score for NF1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>score</th>\n",
       "      <th>gene</th>\n",
       "      <th>histology</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.909091</td>\n",
       "      <td>NF1</td>\n",
       "      <td>neurofibroma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.222222</td>\n",
       "      <td>SAMD4A</td>\n",
       "      <td>malignant_peripheral_nerve_sheath_tumour</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.222222</td>\n",
       "      <td>NF1</td>\n",
       "      <td>malignant_peripheral_nerve_sheath_tumour</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.222222</td>\n",
       "      <td>EHBP1</td>\n",
       "      <td>malignant_peripheral_nerve_sheath_tumour</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.152838</td>\n",
       "      <td>NF1</td>\n",
       "      <td>pheochromocytoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.046005</td>\n",
       "      <td>NF1</td>\n",
       "      <td>neuroblastoma</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      score    gene                                 histology\n",
       "0  0.909091     NF1                              neurofibroma\n",
       "1  0.222222  SAMD4A  malignant_peripheral_nerve_sheath_tumour\n",
       "2  0.222222     NF1  malignant_peripheral_nerve_sheath_tumour\n",
       "3  0.222222   EHBP1  malignant_peripheral_nerve_sheath_tumour\n",
       "4  0.152838     NF1                          pheochromocytoma\n",
       "5  0.046005     NF1                             neuroblastoma"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame(rows, columns=['score', 'gene', 'histology'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Multiple NF1 mutation sites are associated with each histology"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "metadata": {},
   "outputs": [],
   "source": [
    "nf1_mutations={histology:[x for x in W[histology]['NF1']] for histology in nf1_histologies}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>histology</th>\n",
       "      <th>n_mutations</th>\n",
       "      <th>mutations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>pheochromocytoma</td>\n",
       "      <td>22</td>\n",
       "      <td>c.1398del,c.1400C&gt;T,c.1448A&gt;G,c.1466A&gt;G,c.1658...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>neuroblastoma</td>\n",
       "      <td>17</td>\n",
       "      <td>c.*2307C&gt;A,c.1039C&gt;T,c.1063-58G&gt;A,c.1977G&gt;T,c....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>malignant_peripheral_nerve_sheath_tumour</td>\n",
       "      <td>2</td>\n",
       "      <td>c.6642-17G&gt;A,c.6705-17G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>neurofibroma</td>\n",
       "      <td>10</td>\n",
       "      <td>c.1324_1331del,c.3043del,c.5565_5567del,c.5628...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  histology  n_mutations  \\\n",
       "0                          pheochromocytoma           22   \n",
       "1                             neuroblastoma           17   \n",
       "2  malignant_peripheral_nerve_sheath_tumour            2   \n",
       "3                              neurofibroma           10   \n",
       "\n",
       "                                           mutations  \n",
       "0  c.1398del,c.1400C>T,c.1448A>G,c.1466A>G,c.1658...  \n",
       "1  c.*2307C>A,c.1039C>T,c.1063-58G>A,c.1977G>T,c....  \n",
       "2                          c.6642-17G>A,c.6705-17G>A  \n",
       "3  c.1324_1331del,c.3043del,c.5565_5567del,c.5628...  "
      ]
     },
     "execution_count": 246,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame([(x, len(nf1_mutations[x]), ','.join(sorted(nf1_mutations[x]))) for x in nf1_mutations],\n",
    "               columns=['histology', 'n_mutations', 'mutations'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###  NF1 mutation sites are each observed for only a single histology"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [],
   "source": [
    "intersections=[]\n",
    "for i in range(n_nf1_histologies):\n",
    "    hist_i=nf1_histologies[i]\n",
    "    W_i=nf1_mutations[hist_i]\n",
    "    for j in range(i+1,n_nf1_histologies):\n",
    "        hist_j=nf1_histologies[j]\n",
    "        W_j=nf1_mutations[hist_j]\n",
    "        intersection=[x for x in W_i if x in W_j]\n",
    "        intersections.append((hist_i,len(W_i), hist_j, len(W_j), len(intersection)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hist A</th>\n",
       "      <th># Mutations A</th>\n",
       "      <th>hist B</th>\n",
       "      <th># MutationsB</th>\n",
       "      <th># Intersections</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>pheochromocytoma</td>\n",
       "      <td>22</td>\n",
       "      <td>neuroblastoma</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>pheochromocytoma</td>\n",
       "      <td>22</td>\n",
       "      <td>malignant_peripheral_nerve_sheath_tumour</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>pheochromocytoma</td>\n",
       "      <td>22</td>\n",
       "      <td>neurofibroma</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>neuroblastoma</td>\n",
       "      <td>17</td>\n",
       "      <td>malignant_peripheral_nerve_sheath_tumour</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>neuroblastoma</td>\n",
       "      <td>17</td>\n",
       "      <td>neurofibroma</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>malignant_peripheral_nerve_sheath_tumour</td>\n",
       "      <td>2</td>\n",
       "      <td>neurofibroma</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     hist A  # Mutations A  \\\n",
       "0                          pheochromocytoma             22   \n",
       "1                          pheochromocytoma             22   \n",
       "2                          pheochromocytoma             22   \n",
       "3                             neuroblastoma             17   \n",
       "4                             neuroblastoma             17   \n",
       "5  malignant_peripheral_nerve_sheath_tumour              2   \n",
       "\n",
       "                                     hist B  # MutationsB  # Intersections  \n",
       "0                             neuroblastoma            17                0  \n",
       "1  malignant_peripheral_nerve_sheath_tumour             2                0  \n",
       "2                              neurofibroma            10                0  \n",
       "3  malignant_peripheral_nerve_sheath_tumour             2                0  \n",
       "4                              neurofibroma            10                0  \n",
       "5                              neurofibroma            10                0  "
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame(intersections, columns=['hist A', '# Mutations A', 'hist B', '# MutationsB', '# Intersections'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Information from COSMIC and NCBI on each NF1 mutation associated with a histology"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For the NF1 mutation sites associated with each histology, we manually gather and present the variant type, affected pathways and clinical significance."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {},
   "outputs": [],
   "source": [
    "cosmic=pd.read_csv(f'{data_dir}/CosmicGenomeScreensMutantExport.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "metadata": {},
   "outputs": [],
   "source": [
    "def nf1_hist_mutation_info(hist):\n",
    "    global cosmic, nf1_mutations\n",
    "    L=nf1_mutations[hist]\n",
    "    df = cosmic[cosmic['Primary histology']==hist]\n",
    "    df = df[df['Mutation CDS'].isin(L)]\n",
    "    df = df[['Mutation CDS',\n",
    "       'Mutation AA', 'Mutation Description', 'Mutation zygosity', \n",
    "       'Mutation strand', 'SNP',\n",
    "       'FATHMM prediction', 'FATHMM score',\n",
    "       'Pubmed_PMID']]\n",
    "    return df.drop_duplicates()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### malignant_peripheral_nerve_sheath_tumour"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mutation CDS</th>\n",
       "      <th>Mutation AA</th>\n",
       "      <th>Mutation Description</th>\n",
       "      <th>Mutation zygosity</th>\n",
       "      <th>Mutation strand</th>\n",
       "      <th>SNP</th>\n",
       "      <th>FATHMM prediction</th>\n",
       "      <th>FATHMM score</th>\n",
       "      <th>Pubmed_PMID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>17392781</th>\n",
       "      <td>c.6705-17G&gt;A</td>\n",
       "      <td>p.?</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>0.43648</td>\n",
       "      <td>26432421.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28568308</th>\n",
       "      <td>c.6642-17G&gt;A</td>\n",
       "      <td>p.?</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>0.43648</td>\n",
       "      <td>26432421.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Mutation CDS Mutation AA Mutation Description Mutation zygosity  \\\n",
       "17392781  c.6705-17G>A         p.?              Unknown               NaN   \n",
       "28568308  c.6642-17G>A         p.?              Unknown               NaN   \n",
       "\n",
       "         Mutation strand SNP FATHMM prediction  FATHMM score  Pubmed_PMID  \n",
       "17392781               +   n           NEUTRAL       0.43648   26432421.0  \n",
       "28568308               +   n           NEUTRAL       0.43648   26432421.0  "
      ]
     },
     "execution_count": 254,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nf1_hist_mutation_info('malignant_peripheral_nerve_sheath_tumour')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### pheochromocytoma"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mutation CDS</th>\n",
       "      <th>Mutation AA</th>\n",
       "      <th>Mutation Description</th>\n",
       "      <th>Mutation zygosity</th>\n",
       "      <th>Mutation strand</th>\n",
       "      <th>SNP</th>\n",
       "      <th>FATHMM prediction</th>\n",
       "      <th>FATHMM score</th>\n",
       "      <th>Pubmed_PMID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>51506</th>\n",
       "      <td>c.1448A&gt;G</td>\n",
       "      <td>p.D483G</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.98235</td>\n",
       "      <td>23781326.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51508</th>\n",
       "      <td>c.2474C&gt;G</td>\n",
       "      <td>p.S825C</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.98167</td>\n",
       "      <td>23781326.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>607775</th>\n",
       "      <td>c.1398del</td>\n",
       "      <td>p.T467Hfs*6</td>\n",
       "      <td>Deletion - Frameshift</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25545346.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>724772</th>\n",
       "      <td>c.663G&gt;A</td>\n",
       "      <td>p.W221*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99127</td>\n",
       "      <td>25545346.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>902327</th>\n",
       "      <td>c.1400C&gt;T</td>\n",
       "      <td>p.T467I</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.97282</td>\n",
       "      <td>23781326.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>936893</th>\n",
       "      <td>c.1658A&gt;G</td>\n",
       "      <td>p.H553R</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99034</td>\n",
       "      <td>23781326.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2174733</th>\n",
       "      <td>c.2422del</td>\n",
       "      <td>p.L808Ffs*13</td>\n",
       "      <td>Deletion - Frameshift</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23781326.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2174745</th>\n",
       "      <td>c.6705-1G&gt;T</td>\n",
       "      <td>p.?</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99689</td>\n",
       "      <td>25545346.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2908333</th>\n",
       "      <td>c.6522_6523del</td>\n",
       "      <td>p.E2174Dfs*46</td>\n",
       "      <td>Deletion - Frameshift</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25545346.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4121999</th>\n",
       "      <td>c.1466A&gt;G</td>\n",
       "      <td>p.Y489C</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.96016</td>\n",
       "      <td>23781326.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6802594</th>\n",
       "      <td>c.6642-1G&gt;T</td>\n",
       "      <td>p.?</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99689</td>\n",
       "      <td>25545346.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8898545</th>\n",
       "      <td>c.1756_1759del</td>\n",
       "      <td>p.T586Vfs*18</td>\n",
       "      <td>Deletion - Frameshift</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25545346.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9606173</th>\n",
       "      <td>c.1700T&gt;G</td>\n",
       "      <td>p.V567G</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.98640</td>\n",
       "      <td>23781326.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9631463</th>\n",
       "      <td>c.7237C&gt;T</td>\n",
       "      <td>p.Q2413*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.95597</td>\n",
       "      <td>25545346.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10515948</th>\n",
       "      <td>c.2461A&gt;C</td>\n",
       "      <td>p.S821R</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99648</td>\n",
       "      <td>23781326.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11907911</th>\n",
       "      <td>c.2695T&gt;G</td>\n",
       "      <td>p.L899V</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.88144</td>\n",
       "      <td>23781326.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12438223</th>\n",
       "      <td>c.910C&gt;T</td>\n",
       "      <td>p.R304*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.85410</td>\n",
       "      <td>23781326.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22557352</th>\n",
       "      <td>c.243G&gt;A</td>\n",
       "      <td>p.W81*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99127</td>\n",
       "      <td>25545346.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23232768</th>\n",
       "      <td>c.7300C&gt;T</td>\n",
       "      <td>p.Q2434*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.95597</td>\n",
       "      <td>25545346.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24827521</th>\n",
       "      <td>c.4436del</td>\n",
       "      <td>p.F1479Sfs*4</td>\n",
       "      <td>Deletion - Frameshift</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25545346.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31903293</th>\n",
       "      <td>c.6585_6586del</td>\n",
       "      <td>p.E2195Dfs*46</td>\n",
       "      <td>Deletion - Frameshift</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25545346.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33697444</th>\n",
       "      <td>c.4373del</td>\n",
       "      <td>p.F1458Sfs*4</td>\n",
       "      <td>Deletion - Frameshift</td>\n",
       "      <td>het</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25545346.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Mutation CDS    Mutation AA     Mutation Description  \\\n",
       "51506          c.1448A>G        p.D483G  Substitution - Missense   \n",
       "51508          c.2474C>G        p.S825C  Substitution - Missense   \n",
       "607775         c.1398del    p.T467Hfs*6    Deletion - Frameshift   \n",
       "724772          c.663G>A        p.W221*  Substitution - Nonsense   \n",
       "902327         c.1400C>T        p.T467I  Substitution - Missense   \n",
       "936893         c.1658A>G        p.H553R  Substitution - Missense   \n",
       "2174733        c.2422del   p.L808Ffs*13    Deletion - Frameshift   \n",
       "2174745      c.6705-1G>T            p.?                  Unknown   \n",
       "2908333   c.6522_6523del  p.E2174Dfs*46    Deletion - Frameshift   \n",
       "4121999        c.1466A>G        p.Y489C  Substitution - Missense   \n",
       "6802594      c.6642-1G>T            p.?                  Unknown   \n",
       "8898545   c.1756_1759del   p.T586Vfs*18    Deletion - Frameshift   \n",
       "9606173        c.1700T>G        p.V567G  Substitution - Missense   \n",
       "9631463        c.7237C>T       p.Q2413*  Substitution - Nonsense   \n",
       "10515948       c.2461A>C        p.S821R  Substitution - Missense   \n",
       "11907911       c.2695T>G        p.L899V  Substitution - Missense   \n",
       "12438223        c.910C>T        p.R304*  Substitution - Nonsense   \n",
       "22557352        c.243G>A         p.W81*  Substitution - Nonsense   \n",
       "23232768       c.7300C>T       p.Q2434*  Substitution - Nonsense   \n",
       "24827521       c.4436del   p.F1479Sfs*4    Deletion - Frameshift   \n",
       "31903293  c.6585_6586del  p.E2195Dfs*46    Deletion - Frameshift   \n",
       "33697444       c.4373del   p.F1458Sfs*4    Deletion - Frameshift   \n",
       "\n",
       "         Mutation zygosity Mutation strand SNP FATHMM prediction  \\\n",
       "51506                  het               +   n        PATHOGENIC   \n",
       "51508                  het               +   n        PATHOGENIC   \n",
       "607775                 het               +   n               NaN   \n",
       "724772                 het               +   n        PATHOGENIC   \n",
       "902327                 het               +   n        PATHOGENIC   \n",
       "936893                 het               +   n        PATHOGENIC   \n",
       "2174733                het               +   n               NaN   \n",
       "2174745                het               +   n        PATHOGENIC   \n",
       "2908333                NaN               +   n               NaN   \n",
       "4121999                het               +   n        PATHOGENIC   \n",
       "6802594                het               +   n        PATHOGENIC   \n",
       "8898545                het               +   n               NaN   \n",
       "9606173                het               +   n        PATHOGENIC   \n",
       "9631463                het               +   n        PATHOGENIC   \n",
       "10515948               het               +   n        PATHOGENIC   \n",
       "11907911               het               +   n        PATHOGENIC   \n",
       "12438223               het               +   n        PATHOGENIC   \n",
       "22557352               het               +   n        PATHOGENIC   \n",
       "23232768               het               +   n        PATHOGENIC   \n",
       "24827521               het               +   n               NaN   \n",
       "31903293               NaN               +   n               NaN   \n",
       "33697444               het               +   n               NaN   \n",
       "\n",
       "          FATHMM score  Pubmed_PMID  \n",
       "51506          0.98235   23781326.0  \n",
       "51508          0.98167   23781326.0  \n",
       "607775             NaN   25545346.0  \n",
       "724772         0.99127   25545346.0  \n",
       "902327         0.97282   23781326.0  \n",
       "936893         0.99034   23781326.0  \n",
       "2174733            NaN   23781326.0  \n",
       "2174745        0.99689   25545346.0  \n",
       "2908333            NaN   25545346.0  \n",
       "4121999        0.96016   23781326.0  \n",
       "6802594        0.99689   25545346.0  \n",
       "8898545            NaN   25545346.0  \n",
       "9606173        0.98640   23781326.0  \n",
       "9631463        0.95597   25545346.0  \n",
       "10515948       0.99648   23781326.0  \n",
       "11907911       0.88144   23781326.0  \n",
       "12438223       0.85410   23781326.0  \n",
       "22557352       0.99127   25545346.0  \n",
       "23232768       0.95597   25545346.0  \n",
       "24827521           NaN   25545346.0  \n",
       "31903293           NaN   25545346.0  \n",
       "33697444           NaN   25545346.0  "
      ]
     },
     "execution_count": 255,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nf1_hist_mutation_info('pheochromocytoma')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### neuroblastoma"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 256,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mutation CDS</th>\n",
       "      <th>Mutation AA</th>\n",
       "      <th>Mutation Description</th>\n",
       "      <th>Mutation zygosity</th>\n",
       "      <th>Mutation strand</th>\n",
       "      <th>SNP</th>\n",
       "      <th>FATHMM prediction</th>\n",
       "      <th>FATHMM score</th>\n",
       "      <th>Pubmed_PMID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4147230</th>\n",
       "      <td>c.1063-58G&gt;A</td>\n",
       "      <td>p.?</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>0.08494</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4475582</th>\n",
       "      <td>c.6691G&gt;T</td>\n",
       "      <td>p.E2231*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99796</td>\n",
       "      <td>23334666.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5184719</th>\n",
       "      <td>c.2001+39G&gt;T</td>\n",
       "      <td>p.?</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>0.15611</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6271437</th>\n",
       "      <td>c.*2307C&gt;A</td>\n",
       "      <td>p.?</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>0.03680</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8746460</th>\n",
       "      <td>c.1039C&gt;T</td>\n",
       "      <td>p.Q347*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.97489</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10515955</th>\n",
       "      <td>c.1977G&gt;T</td>\n",
       "      <td>p.R659=</td>\n",
       "      <td>Substitution - coding silent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.84177</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11578764</th>\n",
       "      <td>c.910C&gt;G</td>\n",
       "      <td>p.R304G</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.87910</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11907901</th>\n",
       "      <td>c.4836-20659G&gt;T</td>\n",
       "      <td>p.?</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>0.11841</td>\n",
       "      <td>23334666.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11932998</th>\n",
       "      <td>c.6628G&gt;T</td>\n",
       "      <td>p.E2210*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99796</td>\n",
       "      <td>23334666.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12614951</th>\n",
       "      <td>c.7501G&gt;T</td>\n",
       "      <td>p.E2501*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99166</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14232023</th>\n",
       "      <td>c.7126+2T&gt;C</td>\n",
       "      <td>p.?</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99429</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15386927</th>\n",
       "      <td>c.1039C&gt;T</td>\n",
       "      <td>p.P347S</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>0.25960</td>\n",
       "      <td>23334666.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17949560</th>\n",
       "      <td>c.7438G&gt;T</td>\n",
       "      <td>p.E2480*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99166</td>\n",
       "      <td>23334666.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18808644</th>\n",
       "      <td>c.7501G&gt;T</td>\n",
       "      <td>p.E2501*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99166</td>\n",
       "      <td>23334666.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21133830</th>\n",
       "      <td>c.7285C&gt;T</td>\n",
       "      <td>p.R2429*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.90666</td>\n",
       "      <td>23334666.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23587429</th>\n",
       "      <td>c.1978A&gt;T</td>\n",
       "      <td>p.K660*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.95112</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25535086</th>\n",
       "      <td>c.7189+2T&gt;C</td>\n",
       "      <td>p.?</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99429</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27682680</th>\n",
       "      <td>c.4773-20659G&gt;T</td>\n",
       "      <td>p.?</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>0.11841</td>\n",
       "      <td>23334666.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33874733</th>\n",
       "      <td>c.7438G&gt;T</td>\n",
       "      <td>p.E2480*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99166</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34381775</th>\n",
       "      <td>c.7348C&gt;T</td>\n",
       "      <td>p.R2450*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.90666</td>\n",
       "      <td>23334666.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Mutation CDS Mutation AA          Mutation Description  \\\n",
       "4147230      c.1063-58G>A         p.?                       Unknown   \n",
       "4475582         c.6691G>T    p.E2231*       Substitution - Nonsense   \n",
       "5184719      c.2001+39G>T         p.?                       Unknown   \n",
       "6271437        c.*2307C>A         p.?                       Unknown   \n",
       "8746460         c.1039C>T     p.Q347*       Substitution - Nonsense   \n",
       "10515955        c.1977G>T     p.R659=  Substitution - coding silent   \n",
       "11578764         c.910C>G     p.R304G       Substitution - Missense   \n",
       "11907901  c.4836-20659G>T         p.?                       Unknown   \n",
       "11932998        c.6628G>T    p.E2210*       Substitution - Nonsense   \n",
       "12614951        c.7501G>T    p.E2501*       Substitution - Nonsense   \n",
       "14232023      c.7126+2T>C         p.?                       Unknown   \n",
       "15386927        c.1039C>T     p.P347S       Substitution - Missense   \n",
       "17949560        c.7438G>T    p.E2480*       Substitution - Nonsense   \n",
       "18808644        c.7501G>T    p.E2501*       Substitution - Nonsense   \n",
       "21133830        c.7285C>T    p.R2429*       Substitution - Nonsense   \n",
       "23587429        c.1978A>T     p.K660*       Substitution - Nonsense   \n",
       "25535086      c.7189+2T>C         p.?                       Unknown   \n",
       "27682680  c.4773-20659G>T         p.?                       Unknown   \n",
       "33874733        c.7438G>T    p.E2480*       Substitution - Nonsense   \n",
       "34381775        c.7348C>T    p.R2450*       Substitution - Nonsense   \n",
       "\n",
       "         Mutation zygosity Mutation strand SNP FATHMM prediction  \\\n",
       "4147230                NaN               +   n           NEUTRAL   \n",
       "4475582                NaN               +   n        PATHOGENIC   \n",
       "5184719                NaN               +   n           NEUTRAL   \n",
       "6271437                NaN               +   n           NEUTRAL   \n",
       "8746460                NaN               +   n        PATHOGENIC   \n",
       "10515955               NaN               +   n        PATHOGENIC   \n",
       "11578764               NaN               +   n        PATHOGENIC   \n",
       "11907901               NaN               +   n           NEUTRAL   \n",
       "11932998               NaN               +   n        PATHOGENIC   \n",
       "12614951               NaN               +   n        PATHOGENIC   \n",
       "14232023               NaN               +   n        PATHOGENIC   \n",
       "15386927               NaN               +   n           NEUTRAL   \n",
       "17949560               NaN               +   n        PATHOGENIC   \n",
       "18808644               NaN               +   n        PATHOGENIC   \n",
       "21133830               NaN               +   n        PATHOGENIC   \n",
       "23587429               NaN               +   n        PATHOGENIC   \n",
       "25535086               NaN               +   n        PATHOGENIC   \n",
       "27682680               NaN               +   n           NEUTRAL   \n",
       "33874733               NaN               +   n        PATHOGENIC   \n",
       "34381775               NaN               +   n        PATHOGENIC   \n",
       "\n",
       "          FATHMM score  Pubmed_PMID  \n",
       "4147230        0.08494          NaN  \n",
       "4475582        0.99796   23334666.0  \n",
       "5184719        0.15611          NaN  \n",
       "6271437        0.03680          NaN  \n",
       "8746460        0.97489          NaN  \n",
       "10515955       0.84177          NaN  \n",
       "11578764       0.87910          NaN  \n",
       "11907901       0.11841   23334666.0  \n",
       "11932998       0.99796   23334666.0  \n",
       "12614951       0.99166          NaN  \n",
       "14232023       0.99429          NaN  \n",
       "15386927       0.25960   23334666.0  \n",
       "17949560       0.99166   23334666.0  \n",
       "18808644       0.99166   23334666.0  \n",
       "21133830       0.90666   23334666.0  \n",
       "23587429       0.95112          NaN  \n",
       "25535086       0.99429          NaN  \n",
       "27682680       0.11841   23334666.0  \n",
       "33874733       0.99166          NaN  \n",
       "34381775       0.90666   23334666.0  "
      ]
     },
     "execution_count": 256,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nf1_hist_mutation_info('neuroblastoma')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### neurofibroma"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 257,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mutation CDS</th>\n",
       "      <th>Mutation AA</th>\n",
       "      <th>Mutation Description</th>\n",
       "      <th>Mutation zygosity</th>\n",
       "      <th>Mutation strand</th>\n",
       "      <th>SNP</th>\n",
       "      <th>FATHMM prediction</th>\n",
       "      <th>FATHMM score</th>\n",
       "      <th>Pubmed_PMID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4147230</th>\n",
       "      <td>c.1063-58G&gt;A</td>\n",
       "      <td>p.?</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>0.08494</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4475582</th>\n",
       "      <td>c.6691G&gt;T</td>\n",
       "      <td>p.E2231*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99796</td>\n",
       "      <td>23334666.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5184719</th>\n",
       "      <td>c.2001+39G&gt;T</td>\n",
       "      <td>p.?</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>0.15611</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6271437</th>\n",
       "      <td>c.*2307C&gt;A</td>\n",
       "      <td>p.?</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>0.03680</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8746460</th>\n",
       "      <td>c.1039C&gt;T</td>\n",
       "      <td>p.Q347*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.97489</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10515955</th>\n",
       "      <td>c.1977G&gt;T</td>\n",
       "      <td>p.R659=</td>\n",
       "      <td>Substitution - coding silent</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.84177</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11578764</th>\n",
       "      <td>c.910C&gt;G</td>\n",
       "      <td>p.R304G</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.87910</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11907901</th>\n",
       "      <td>c.4836-20659G&gt;T</td>\n",
       "      <td>p.?</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>0.11841</td>\n",
       "      <td>23334666.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11932998</th>\n",
       "      <td>c.6628G&gt;T</td>\n",
       "      <td>p.E2210*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99796</td>\n",
       "      <td>23334666.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12614951</th>\n",
       "      <td>c.7501G&gt;T</td>\n",
       "      <td>p.E2501*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99166</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14232023</th>\n",
       "      <td>c.7126+2T&gt;C</td>\n",
       "      <td>p.?</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99429</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15386927</th>\n",
       "      <td>c.1039C&gt;T</td>\n",
       "      <td>p.P347S</td>\n",
       "      <td>Substitution - Missense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>0.25960</td>\n",
       "      <td>23334666.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17949560</th>\n",
       "      <td>c.7438G&gt;T</td>\n",
       "      <td>p.E2480*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99166</td>\n",
       "      <td>23334666.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18808644</th>\n",
       "      <td>c.7501G&gt;T</td>\n",
       "      <td>p.E2501*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99166</td>\n",
       "      <td>23334666.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21133830</th>\n",
       "      <td>c.7285C&gt;T</td>\n",
       "      <td>p.R2429*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.90666</td>\n",
       "      <td>23334666.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23587429</th>\n",
       "      <td>c.1978A&gt;T</td>\n",
       "      <td>p.K660*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.95112</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25535086</th>\n",
       "      <td>c.7189+2T&gt;C</td>\n",
       "      <td>p.?</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99429</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27682680</th>\n",
       "      <td>c.4773-20659G&gt;T</td>\n",
       "      <td>p.?</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>NEUTRAL</td>\n",
       "      <td>0.11841</td>\n",
       "      <td>23334666.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33874733</th>\n",
       "      <td>c.7438G&gt;T</td>\n",
       "      <td>p.E2480*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.99166</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34381775</th>\n",
       "      <td>c.7348C&gt;T</td>\n",
       "      <td>p.R2450*</td>\n",
       "      <td>Substitution - Nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>+</td>\n",
       "      <td>n</td>\n",
       "      <td>PATHOGENIC</td>\n",
       "      <td>0.90666</td>\n",
       "      <td>23334666.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Mutation CDS Mutation AA          Mutation Description  \\\n",
       "4147230      c.1063-58G>A         p.?                       Unknown   \n",
       "4475582         c.6691G>T    p.E2231*       Substitution - Nonsense   \n",
       "5184719      c.2001+39G>T         p.?                       Unknown   \n",
       "6271437        c.*2307C>A         p.?                       Unknown   \n",
       "8746460         c.1039C>T     p.Q347*       Substitution - Nonsense   \n",
       "10515955        c.1977G>T     p.R659=  Substitution - coding silent   \n",
       "11578764         c.910C>G     p.R304G       Substitution - Missense   \n",
       "11907901  c.4836-20659G>T         p.?                       Unknown   \n",
       "11932998        c.6628G>T    p.E2210*       Substitution - Nonsense   \n",
       "12614951        c.7501G>T    p.E2501*       Substitution - Nonsense   \n",
       "14232023      c.7126+2T>C         p.?                       Unknown   \n",
       "15386927        c.1039C>T     p.P347S       Substitution - Missense   \n",
       "17949560        c.7438G>T    p.E2480*       Substitution - Nonsense   \n",
       "18808644        c.7501G>T    p.E2501*       Substitution - Nonsense   \n",
       "21133830        c.7285C>T    p.R2429*       Substitution - Nonsense   \n",
       "23587429        c.1978A>T     p.K660*       Substitution - Nonsense   \n",
       "25535086      c.7189+2T>C         p.?                       Unknown   \n",
       "27682680  c.4773-20659G>T         p.?                       Unknown   \n",
       "33874733        c.7438G>T    p.E2480*       Substitution - Nonsense   \n",
       "34381775        c.7348C>T    p.R2450*       Substitution - Nonsense   \n",
       "\n",
       "         Mutation zygosity Mutation strand SNP FATHMM prediction  \\\n",
       "4147230                NaN               +   n           NEUTRAL   \n",
       "4475582                NaN               +   n        PATHOGENIC   \n",
       "5184719                NaN               +   n           NEUTRAL   \n",
       "6271437                NaN               +   n           NEUTRAL   \n",
       "8746460                NaN               +   n        PATHOGENIC   \n",
       "10515955               NaN               +   n        PATHOGENIC   \n",
       "11578764               NaN               +   n        PATHOGENIC   \n",
       "11907901               NaN               +   n           NEUTRAL   \n",
       "11932998               NaN               +   n        PATHOGENIC   \n",
       "12614951               NaN               +   n        PATHOGENIC   \n",
       "14232023               NaN               +   n        PATHOGENIC   \n",
       "15386927               NaN               +   n           NEUTRAL   \n",
       "17949560               NaN               +   n        PATHOGENIC   \n",
       "18808644               NaN               +   n        PATHOGENIC   \n",
       "21133830               NaN               +   n        PATHOGENIC   \n",
       "23587429               NaN               +   n        PATHOGENIC   \n",
       "25535086               NaN               +   n        PATHOGENIC   \n",
       "27682680               NaN               +   n           NEUTRAL   \n",
       "33874733               NaN               +   n        PATHOGENIC   \n",
       "34381775               NaN               +   n        PATHOGENIC   \n",
       "\n",
       "          FATHMM score  Pubmed_PMID  \n",
       "4147230        0.08494          NaN  \n",
       "4475582        0.99796   23334666.0  \n",
       "5184719        0.15611          NaN  \n",
       "6271437        0.03680          NaN  \n",
       "8746460        0.97489          NaN  \n",
       "10515955       0.84177          NaN  \n",
       "11578764       0.87910          NaN  \n",
       "11907901       0.11841   23334666.0  \n",
       "11932998       0.99796   23334666.0  \n",
       "12614951       0.99166          NaN  \n",
       "14232023       0.99429          NaN  \n",
       "15386927       0.25960   23334666.0  \n",
       "17949560       0.99166   23334666.0  \n",
       "18808644       0.99166   23334666.0  \n",
       "21133830       0.90666   23334666.0  \n",
       "23587429       0.95112          NaN  \n",
       "25535086       0.99429          NaN  \n",
       "27682680       0.11841   23334666.0  \n",
       "33874733       0.99166          NaN  \n",
       "34381775       0.90666   23334666.0  "
      ]
     },
     "execution_count": 257,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nf1_hist_mutation_info('neuroblastoma')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "octf",
   "language": "python",
   "name": "octf"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
